src/deserializer.rs

```diff
@@ -738,17 +734,17 @@ impl DeserializeError {
 }
 
 impl DeserializeErrorKind {
-    fn description(&self) -> &str {
+    fn description(&self) -> &dyn core::fmt::Display {
         use self::DeserializeErrorKind::*;
 
         match *self {
-            Message(_) => "deserialization error",
-            Unsupported(_) => "unsupported deserializer method",
-            UnexpectedEndOfRow => "expected field, but got end of row",
-            InvalidUtf8(ref err) => err.description(),
-            ParseBool(ref err) => err.description(),
-            ParseInt(ref err) => err.description(),
-            ParseFloat(ref err) => err.description(),
+            Message(_) => &"deserialization error",
+            Unsupported(_) => &"unsupported deserializer method",
+            UnexpectedEndOfRow => &"expected field, but got end of row",
+            InvalidUtf8(ref err) => err,
+            ParseBool(ref err) => err,
+            ParseInt(ref err) => err,
+            ParseFloat(ref err) => err,
         }
     }
 }
```

Comment:
I removed this in an edit... not sure how it returned. The Display impl calls this function only on the `UnexpectedEndOfRow` variant, so that may as well skip the indirection.